### ISSUE TITLE
Optional ConfigFilePropertySource

### ISSUE THREAD

**Issue by Globegitter** [sksamuel/hoplite#76]
Hi, I just found this library and so far it is really looking good. But one thing I noticed we are missing for internal adoption is the concept of an optional `ConfigFilePropertySource`, i.e. given a config setup of:

```
val config = ConfigLoader()
        .withParameterMapper(SnakeCaseParamMapper)
        .withPropertySource(ConfigFilePropertySource(
            FileSource.ClasspathSource("/local.properties"),
            defaultParserRegistry()
        ))
        .loadConfigOrThrow<Config>("/config.properties")
```

I want the config to be loaded and parsed when `local.properties` does not exist or is empty but `config.properties` has all the necessary config. Reason for that being is that we have a default config file which has all non-personal, non-secret config values checked-in and the local.config is a file on gitignore meant to be able to override any existing values or add any of the missing ones. But for some of our applications all of our config values can be checked-in and so a local.properties file is not strictly needed (you might still choose to override some anyway). 

Would it be possible to have a `withOptionalPropertySource` method or similar to allow for this use case?

**Comment by sksamuel** [comment]
Seems like a good use case.

For now, you could use the UserSettingsPropertySource which is hardcoded.

**Comment by sksamuel** [comment]
I've released 1.2.0 which I've added this:

```kotlin
    ConfigLoader()
      .withPropertySource(ConfigFilePropertySource.optionalResource("/missing.yml"))
      .loadConfig<Foo>("/basic.yml")
```

which is sugar for

```kotlin
val config = ConfigLoader()
        .withParameterMapper(SnakeCaseParamMapper)
        .withPropertySource(ConfigFilePropertySource(
            FileSource.ClasspathSource("/local.properties"),
            defaultParserRegistry(),
            true // optional flag
        ))
        .loadConfigOrThrow<Config>("/config.properties")
```

Also the SnakeCaseParamMapper is registered by default FYI.

**Comment by Globegitter** [comment]
Awesome! Will try it out.
